pkg/vpcmodel/nodesExplainability.go

```diff
@@ -0,0 +1,151 @@
+	for _, node := range c.Nodes {
+		// currently, supported: network interface given takes only that one.
+		//  todo:   if address not given but only vsi name - take all network interfaces of that vsi
+		if name == node.Name() {
+			return node
+		}
+	}
+	return nil
+}
+
+// ExplainConnectivity todo: this will not be needed here once we connect explanbility to the cli
+func (c *VPCConfig) ExplainConnectivity(srcName, dstName string) (explanation string, err error) {
+	src := c.getVsiNode(srcName)
+	if src == nil {
+		return "", fmt.Errorf("src %v does not represent a VSI", srcName)
+	}
+	dst := c.getVsiNode(dstName)
+	if dst == nil {
+		return "", fmt.Errorf("dst %v does not represent a VSI", dstName)
+	}
+	rulesOfConnection, err1 := c.GetRulesOfConnection(src, dst)
+	if err1 != nil {
+		return "", err1
+	}
+	return rulesOfConnection.String(src, dst, c), nil
+}
+
+func (c *VPCConfig) getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(
+	src, dst Node, isIngress bool, layer string) (rules *[]RulesInFilter, err error) {
+	filter := c.getFilterTrafficResourceOfKind(layer)
+	if filter == nil {
+		return nil, fmt.Errorf("layer %v not found in configuration", layer)
+	}
+	rulesOfFilter, err := filter.RulesInConnectivity(src, dst, isIngress)
+	if err != nil {
+		return nil, err
+	}
+	return &rulesOfFilter, nil
+}
+
+func (c *VPCConfig) GetRulesOfConnection(src, dst Node) (rulesOfConnection *RulesOfConnection, err error) {
+	filterLayers := []string{SecurityGroupLayer}
+	rulesOfConnection = &RulesOfConnection{make([]rulesInLayer, len(filterLayers)),
+		make([]rulesInLayer, len(filterLayers))}
+	for i, layer := range filterLayers {
+		// ingress rules
+		ingressRules, err1 := c.getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(src, dst, true, layer)
+		if err1 != nil {
+			return nil, err1
+		}
+		ingressThisLayer := rulesInLayer{layer: layer, rules: *ingressRules}
+		rulesOfConnection.ingressRules[i] = ingressThisLayer
+
+		// egress rules
+		egressRules, err2 := c.getFiltersEnablingRulesBetweenNodesPerDirectionAndLayer(src, dst, false, layer)
+		if err2 != nil {
+			return nil, err2
+		}
+		if len(*egressRules) > 0 {
+			egressThisLayer := rulesInLayer{layer: layer, rules: *egressRules}
+			rulesOfConnection.egressRules[i] = egressThisLayer
+		}
+	}
+	return rulesOfConnection, nil
+}
+
+func (rulesOfConnection *RulesOfConnection) String(src, dst Node, c *VPCConfig) string {
+	noIngressRules := !rulesOfConnection.ingressRules.hasRules()
+	noEgressRules := !rulesOfConnection.egressRules.hasRules()
+	switch {
+	case noIngressRules && noEgressRules:
+		return fmt.Sprintf("No connection between %v and %v; connection blocked both by ingress and egress\n", src.Name(), dst.Name())
+	case noIngressRules:
+		return fmt.Sprintf("No connection between %v and %v; connection blocked by ingress\n", src.Name(), dst.Name())
+	case noEgressRules:
+		return fmt.Sprintf("No connection between %v and %v; connection blocked by egress\n", src.Name(), dst.Name())
```

Comment:
may want to add info about rules enabling one direction?

Reply:
done